assembly/assembly-ide-war/pom.xml

```diff
@@ -347,6 +345,19 @@
                 <artifactId>exec-maven-plugin</artifactId>
                 <executions>
                     <execution>
+                        <id>gwt-xml</id>
+                        <phase>generate-sources</phase>
+                        <goals>
+                            <goal>java</goal>
+                        </goals>
+                        <configuration>
+                            <mainClass>org.eclipse.che.util.CompilingGwtXmlGenerator</mainClass>
```

Comment:
I think it's quite straightforward to build maven plugin.
From end-user point of view, it's also easier to add/manage
```
<plugin>
  <groupId>org.eclipse.che.assembly</groupId>
  <artifactId>gwt-asssembly-plugin</artifactId>
</plugin>
```
instead of 
```
           <plugin>
                <groupId>org.codehaus.mojo</groupId>
                <artifactId>exec-maven-plugin</artifactId>
                <executions>
                    <execution>
                        <id>gwt-xml</id>
                        <phase>generate-sources</phase>
                        <goals>
                            <goal>java</goal>
                        </goals>
                        <configuration>
                            <mainClass>org.eclipse.che.util.CompilingGwtXmlGenerator</mainClass>
                            <arguments>
                                <argument>--rootDir=${generated.sources.directory}</argument>
                            </arguments>
                        </configuration>
                    </execution>
             </plugin>
```

because in maven mojo, the lifecycle and generated properties can be automatically configured/injected



(side note: we could also eventually manage a different maven type like : che-ide-assembly
and all will be automatically configured)

<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
    <modelVersion>4.0.0</modelVersion>
    <parent>
        <artifactId>che-assembly-parent</artifactId>
        <groupId>org.eclipse.che</groupId>
        <version>5.5.0-SNAPSHOT</version>
    </parent>
    <artifactId>assembly-ide-war</artifactId>
    <packaging>che-assembly-ide</packaging>
    <dependencies>
        ...listing only dependencies specific to the app...
   </dependencies>
</project>